File: EduCenter.Domain/Models/Group.cs
namespace EduCenter.Domain.Models;

public class Group : BaseEntity
{
    public int TeacherId { get; set; }
    public int SubjectId { get; set; }
    public bool IsActive { get; set; }
    public int MaxNumberOfClasses { get; set; }
    public int NumberOfClassesLeft { get; set; }
}

File: ClientApp/src/app/models/group.ts
export interface Group {
  id: number;
  name: string;
  teacherId: number;
  teacherFirstName: string;
  teacherLastName: string;
  subjectId: number;
  subjectName: string;
  isActive: boolean;
  maxNumberOfClasses: number;
  numberOfClassesLeft: number;
  createdAt: Date;
  updatedAt: Date;
  studentCount: number;
}
